### PR TITLE
[#1848] Fix limit verification outcomes in AMQP adapter.

### PR DESCRIPTION
**Pull request by kaniyan**
In AMQP adapter, the checks (such as connection limit, adapter enabled or not ) are performed as part of the `AmqpAdapterSaslAuthenticatorFactory`. If the connection limit check has failed or the adapter is disabled, this approach always returned authentication failure instead of propagating the actual reason. In this PR, these checks are moved out of the `AmqpAdapterSaslAuthenticatorFactory` and there by fixing the issue #1848.

### CODE REVIEW

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java [hunk 1]
```diff
@@ -357,18 +357,26 @@ private void processRemoteOpen(final ProtonConnection con) {
...
                 // the SASL handshake will already have authenticated the device
-                // and will have verified that the adapter is enabled for the tenant
-                // we still need to check if the device/gateway exists and is enabled
-                checkDeviceRegistration(authenticatedDevice, span.context())
-                .map(ok -> {
-                    log.debug("{} is registered and enabled", authenticatedDevice);
-                    span.log("device is registered and enabled");
-                    return ok;
-                }).setHandler(connectAuthorizationCheck);
+                // and verify if the adapter is enabled for the tenant
```
Comment (sophokles73) [inline adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java:365]: what about
```
// we still need to verify that
// the adapter is enabled for the tenant,
// the device/gateway exists and is enabled and
// that the connection limit for the tenant and the adapter are not exceeded.
```

File: adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapterTest.java [hunk 2]
```diff
@@ -13,6 +13,7 @@
 package org.eclipse.hono.adapter.amqp.impl;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (sophokles73) [inline adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapterTest.java:16]: can you please refactor the tests to use assertj instead?

Comment (kaniyan) [reply adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapterTest.java:16]:
```
 assertThat(errorConditionCaptor.getValue().getCondition().toString())
                .isEqualTo(AmqpError.UNAUTHORIZED_ACCESS.toString());
```

```
 assertEquals(AmqpError.UNAUTHORIZED_ACCESS, 
                errorConditionCaptor.getValue().getCondition());
```
IMHO in the above context, `asserEquals(...)` suited better as there is no need to explicitly convert to string and then compare it as in the case of `asserThat(...)` as it doesn't allow `Symbol ` as an argument. Also we widely use `assertEquals(...)` in other tests too and hence I used it here too. I could also refactor it to use it to `assertThat(...)`. WDYT?

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java (hunk 1 above)
Comment (kaniyan) [reply adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java]: Done.

**Comment by kaniyan** [comment]
> Can you also please update the AMQP adapter user guide with information regarding its behavior wrt to exceeded limits?

@sophokles73 I have now updated the user guide and also the release notes.

File: adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapterTest.java (hunk 2 above)
Comment (sophokles73) [reply adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapterTest.java:16]: I see. Let's keep it like this, then ...

**Pull request merged**